site/profile/manifests/ccs/file_transfer.pp

```diff
@@ -39,8 +55,132 @@
   String $pkgurl = $profile::ccs::common::pkgurl,
   String $pkgurl_user = $profile::ccs::common::pkgurl_user,
   String $pkgurl_pass = $profile::ccs::common::pkgurl_pass,
+  Boolean $s3daemon = false,
+  Stdlib::HTTPUrl $s3daemon_repo_url = 'https://github.com/lsst-dm/s3daemon',
+  Optional[String] $s3daemon_repo_rev = undef,
+  String $s3daemon_repo_directory = '/home/ccs-ipa/s3daemon/git',
+  String $s3daemon_env_file = '/home/ccs-ipa/s3daemon/env',
+  String $s3daemon_env_url = 'https://s3dfrgw.slac.stanford.edu',
+  String $s3daemon_env_access = 'access-key',
+  String $s3daemon_env_secret = 'secret-key',
 ) {
   if $install {
+    if $s3daemon {
```

Comment:
Probably the install parameter exists because I wanted to reuse some parameters from profile::ccs::common, and thought that meant I should include profile::ccs::file_transfer in common, rather than having it be entirely separate. I then needed a way to control which hosts it was applied on. But I suppose it is reasonable to assume that profile::ccs::common will always be in use, so I could indeed make this separate and get rid of install.

In the short/medium term, I believe s3daemon will replace the old way of doing file transfer, so hopefully it is only a temporary situation to need two separate flags.